glean-core/src/event_database/mod.rs

```diff
@@ -44,16 +53,19 @@ pub struct RecordedEvent {
     pub extra: Option<HashMap<String, String>>,
 }
 
-impl RecordedEvent {
-    /// Serialize an event to JSON, adjusting its timestamp relative to a base timestamp
-    fn serialize_relative(&self, timestamp_offset: u64) -> JsonValue {
-        json!(&RecordedEvent {
-            timestamp: self.timestamp - timestamp_offset,
-            category: self.category.clone(),
-            name: self.name.clone(),
-            extra: self.extra.clone(),
-        })
-    }
+/// Represents the stored data for a single event.
+#[derive(Debug, Clone, Deserialize, Serialize, PartialEq, Eq)]
+struct StoredEvent {
+    #[serde(flatten)]
```

Comment:
A load-bearing flatten!

Reply:
Do we have a test to ensure the serialized form doesn't change?

Reply:
🤔 https://github.com/mozilla/glean/blob/adef1eca3fb1c73a1cb6cf0acc6d6dce63741df0/glean-core/tests/event.rs#L144-L149

Reply:
See `deserialize_existing_data` in event_database/mod.rs